docs/specs/blueberry.md

```diff
@@ -0,0 +1,160 @@
+Once all possibilities of create a block not advancing chain time are exhausted, we attempt to build a block which _may_ advance chain time as follows:
+
+1. If the local clock's time is greater than or equal to the earliest change-event timestamp of the staker set, an advance time transaction is issued into a Proposal Block to move current chain time to the earliest change timestamp of the staker set. Upon this Proposal block's acceptance, chain time will be move ahead and all scheduled changes (e.g. promoting a staker from pending to current) will be carried out.
+2. If the mempool contains any proposal transactions, the mempool proposal transaction with the earliest start time is selected and included into a Proposal Block[^1]. A mempool proposal transaction as is won't change the current chain time[^2]. However there is an edge case to consider: on low activity chains (e.g. Fuji P-chain) chain time may fall significantly behind the local clock. If a proposal transaction is finally issued, its start time is likely to be quite far in the future relative to the current chain time. This would cause the proposal transaction to be considered invalid and rejected, since a staker added by a proposal transaction's start time must be at most 366 hours (two weeks) after current chain time. To avoid this edge case on low-activity chains, an advance time transaction is issued first to move chain time to the local clock's time. As soon as chain time is advanced, the mempool proposal transaction will be issued and accepted.
+
+Note that the order in which these steps are executed matters. A block updating chain time would be deemed invalid if it would advance time beyond the staker set's next change event, skipping the associated changes. The order above ensures this never happens because it checks first if chain time should be moved to the time of the next staker set change. It can also be verified by inspection that the timestamp selected for the advance time transactions always respect the synchrony bound.
+
+Block formation terminates as soon as any of the steps executed manage to select transactions to be included into a block. Otherwise, an error is raised to signal that there are no transactions to be issued. Finally a timer is kicked off to schedule the next block formation attempt.
+
+### Blueberry
+
+#### Blueberry Block Content
+
+Blueberry allows the following block types with the following content:
+
+- _Standard Blocks_ may contain multiple transactions of the following types:
+  - CreateChainTx
+  - CreateSubnetTx,
+  - ImportTx,
+  - ExportTx,
+  - AddValidatorTx,
+  - AddDelegatorTx,
+  - AddSubnetValidatorTx.
+- _Proposal Blocks_ may contain a single transaction of the following types:
+  - RewardValidatorTx,
+- _Options blocks_, i.e. _Commit Block_ and _Abort Block_ do not contain any transactions.
+
+Note that each block has an header containing:
+
+- ParentID
+- Height
+- Time
+
+So the two main differences with respect to Apricot are:
```

Comment:
There is a thrid difference missing: AdvanceTimeTx transactions is not accepted anymore (for reasons explained in section `Chain Time Update Mechanism`